### ISSUE TITLE
SonarQube  - ERROR: Failed to parse json stylelint report

### ISSUE THREAD

**Issue by HarveyHuiZhang** [SonarSource/sonar-css#161]
Environment:
SonarQube server version: Enterprise EditionVersion 7.6 (build 21501)LGPL v3Community
SonarQube client version: sonar-scanner-cli-3.2.0.1227-windows
SonarCSS version on SonarQube version: 1.0.3 (build 724) installed
StyleLint version 9.10.1

Description:
StyleLint report was generated. Below error message occurred, when using SonarQube scanner to scan code and merge generated StyleLint report into SonarQube server.

17:08:04.590 INFO: Sensor SonarCSS Rules [cssfamily] (done) | time=46705ms
17:08:04.590 INFO: Sensor Import of stylelint issues [cssfamily]
17:08:04.592 INFO: Importing C:\Users\sonar\css_bad_example\stylelint-report.json
17:08:04.597 ERROR: Failed to parse json stylelint report
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $
        at com.google.gson.Gson.fromJson(Gson.java:891)
        at com.google.gson.Gson.fromJson(Gson.java:817)
        at org.sonar.css.plugin.StylelintReportSensor.importReport(StylelintReportSensor.java:82)
        at org.sonar.css.plugin.StylelintReportSensor.lambda$execute$1(StylelintReportSensor.java:75)
        at java.util.ArrayList.forEach(ArrayList.java:1249)
        at org.sonar.css.plugin.StylelintReportSensor.execute(StylelintReportSensor.java:75)
        at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:85)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:59)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:77)
        at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:59)
        at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:82)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:408)
        at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:403)
        at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:360)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:126)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
        at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:73)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
        at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
        at com.sun.proxy.$Proxy0.execute(Unknown Source)
        at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:171)
        at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:128)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:111)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
        at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $
        at com.google.gson.stream.JsonReader.beginArray(JsonReader.java:351)
        at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:70)
        at com.google.gson.Gson.fromJson(Gson.java:879)
        ... 35 common frames omitted

17:08:04.598 INFO: Sensor Import of stylelint issues [cssfamily] (done) | time=8ms
17:08:04.599 INFO: Sensor JavaXmlSensor [java]

**Comment by vilchik-elena** [comment]
Could you tell how you generated your report? Also it will help if you show content of it.

**Comment by HarveyHuiZhang** [comment]
Hi Vilchik,
Per Style official site, I installed Stylelin on my local dev machine.
Then I use below command to generate Stylelint report
node-v10.15.3-win-x64/node_modules/.bin/stylelint "$projectPath/**/*.css" -f json > $projectPath/stylelint-report.json

**Comment by HarveyHuiZhang** [comment]
Below is my stylelint-report.json file content:

[{"source":"C:\\test\\css_bad_examples\\a_color.css","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":true,"warnings":[{"line":4,"column":1,"rule":"block-closing-brace-empty-line-before","severity":"error","text":"Unexpected empty line before closing brace (block-closing-brace-empty-line-before)"},{"line":4,"column":1,"rule":"no-missing-end-of-source-newline","severity":"error","text":"Unexpected missing end-of-source newline (no-missing-end-of-source-newline)"}]}]

**Comment by vilchik-elena** [comment]
@HarveyHuiZhang can you provide file itself? I suspect there are some invisible characters

**Comment by HarveyHuiZhang** [comment]
@vilchik-elena I uploaded these original files, please refer from [css_bad_examples](https://github.com/HarveyHuiZhang/css_bad_examples.git)

**Comment by vilchik-elena** [comment]
Indeed you have a BOM character in this file. I've created a ticket to support files with BOM #164. 
In meantime you can remove it manually or change encoding.

**Comment by vilchik-elena** [comment]
In fact problem in not in BOM but in the encoding of the file (UTF-16-LE in the example)

**Comment by SomethingNew71** [comment]
This is still an issue using 
- sonar-css version 1.2.1 
- sonarqube 8.5.1 
- node 12.18.3
- sonar-scanner 3.1.0

I have tried changing the encoding and line endings of the stylelintrc, I have tried different file names (eg. .stylelintrc, .stylelintrc.json, and .stylelintrc.js). None of these have provided a solution and I am at my wits end trying to get this running in our enterprise applicaiton. 

Any idea why this is still happening?
